lib/projfs.c

```diff
@@ -1527,3 +1523,15 @@ int _projfs_make_dir(struct projfs *fs, const char *path, mode_t mode,
 	return 0;
 }
 
+int projfs_write_file_contents(int fd, const void *bytes, unsigned int count)
+{
+	while (count) {
+		ssize_t res = write(fd, bytes, count);
+		if (res == -1)
+			return errno;
+		bytes = (void *)(((uintptr_t)bytes) + res);
```

Comment:
Normal precedence should, I think, let you just do `(void *)((uintptr_t)bytes + res)`, but I'm also wondering if it's slightly more readable to use `(char*)bytes + res`, which should be equivalent since `sizeof(char) == 1` in C.

Reply:
![This is the power of math, people!](https://user-images.githubusercontent.com/1915/53534368-ae59f980-3b52-11e9-93c3-1b051df88ec3.gif)

Reply:
Just watched s2.e5 _Saints of Imperfection_!  Major spoilers!